src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs

```diff
@@ -89,6 +90,8 @@ public void Start()
         return handler;
     }
 
+    protected virtual bool EmptyQueueUponSerialRequest => false;
```

Comment:
Lets bike shed :)

I don't think "EmptyQueue" is right here. This doesn't actually effect the queue at all, but rather the things that have already been pulled out of it. Also it doesn't just empty things, it cancels then. Finally, "Serial" (and "Parallel") is the verbiage from O#, but not used in CLaSP. I would call this `CancelInProgressWorkUponMutatingRequest`, to be super clear about what it is going to do.